docs/LifeOfAnInstruction.md

```diff
@@ -0,0 +1,202 @@
+# Machine code to bitcode: the life of an instruction
+
+This document describes how machine code instructions are lifted into LLVM bitcode. It should provide a detailed, inside-out overview of how McSema performs binary translation. This document omits some important steps, so consulting the [Navigating the code](NagivatingTheCode.md) document will be helpful.
+
+### Running example
+
+This document will use the instructions in the following basic block as a running example.
+
+![Sample basic block](images/instruction_life_block.png)
+
+## Step 1: CFG protocol buffer representation
+
+The first step to lifting involves getting the machine code instructions of some binary file into a format the McSema understands. We use a common and simple format so that McSema doesn't need to understand the many executable file formats ([ELF](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format), [Mach-O](https://en.wikipedia.org/wiki/Mach-O), [PE](https://en.wikipedia.org/wiki/Portable_Executable), etc.).
+
+The file format used by McSema is a [CFG protocol buffer](/mcsema/CFG/CFG.proto). This file is produced using the [`mcsema-disass`](/mcsema/tools/mcsema_disass) tool. The CFG file for the above code block contains the following data.
+
+```protobuf
+Module {
+  internal_funcs = [
+    Function {
+      entry_address = 0x804b7a3;
+      symbol_name = "sub_804b7a3";
+      local_noreturn = true;
+      blocks = [
+          Block {
+            base_address = 0x804b7a3;
+            block_follows = [ ];
+            insts = [
+
+              // mov    eax, 0x1
+              Instruction {
+                inst_addr = 0x804b7a3;
+                inst_len = 5;
+                inst_bytes = "\xb8\x01\x00\x00\x00";
+              },
+
+              // push   ebx
+              Instruction {
+                inst_addr = 0x0804b7a8;
+                inst_len = 1;
+                inst_bytes = "\x53";
+              },
+
+              // mov    ebx, dword [esp+0x8]
+              Instruction {
+                inst_addr = 0x0804b7a9;
+                inst_len = 4;
+                inst_bytes = "\x8b\x5c\x24\x08";
+              },
+
+              // int    0x80
+              Instruction {
+                inst_addr = 0x0804b7ad;
+                inst_len = 2;
+                inst_bytes = "\xcd\x80";
+              } ];
+          } ];
+    } ];
+  ...
+}      
+```
+
+## Step 2: Lifting to bitcode
```

Comment:
Can you mention how sub_804b7a3 got its name, and how it gets populated? Also link to some of the raw semantics implementations for the instructions in the block (i see you do this later with `int`).